app/models/reservation.rb

```diff
@@ -0,0 +1,26 @@
+class Reservation < ApplicationRecord
+
+  belongs_to :branch
+  belongs_to :customer
+  has_many :line_items, dependent: :destroy, after_add: :update_total
+
+  validates :booking_schedule, presence: true
+
+  before_create :book_tables
+
+  def update_total(line_item)
+    update_column(:total_amount, total_amount + line_item.product.price * line_item.quantity )
+  end
+
+  def book_tables
+    required_tables = person_count/4
```

Comment:
space around operators missing

Reply:
create another model for table and store capacity there

Reply:
Fixed.
Table model is created for tables.
and reservations_tables join table used to store tables booked for each reservation.